cliquet/initialization.py

```diff
@@ -338,11 +338,14 @@ def initialize(config, version=None, project_name=None, default_settings=None):
         in application settings.
     :param dict default_settings: Override cliquet default settings values.
     """
-    if default_settings:
-        config.add_settings(default_settings)
-
     settings = config.get_settings()
 
+    if default_settings:
+        for key, value in default_settings.items():
+            if not settings.get(key):
```

Comment:
If we could manage to read settings in that order that would be much better:

 - cliquet.DEFAULT_SETTINGS
 - kinto.DEFAULT_SETTINGS (updates keys from cliquet.DEFAULT_SETTINGS)
 - read the user config.ini